xen-bugtool

```diff
@@ -1920,7 +1920,7 @@ class TarOutput(ArchiveWithTarSubarchives):
         if output_fd == -1:
             self.tf = tarfile.open(self.filename, mode)
         else:
-            self.tf = tarfile.open(None, 'w|', os.fdopen(output_fd, 'ab'))
+            self.tf = tarfile.open(None, "w|", os.fdopen(output_fd, "wb"))
```

Comment:
If it is a file and not a pipe, it would, yes.

But appending `tar` output to an existing file, would not be a valid tar file.

Yes, if some use case of wanting to append the tar block archive data to an existing file, this would truncate that other file.

I don's see why anyone would want to do that, but I can add a check if the output is a pipe (where we can't seek) or, I can add an exception handler for `os.fdopen(output_fd, "ab")` and fall back to `os.fdopen(output_fd, "ab")` when that is not accepted by the Python interpreter.